qa/rpc-tests/excessive.py

```diff
@@ -12,17 +12,231 @@
+        self.nodes[2].set("net.excessiveBlock=2000000")
+        self.nodes[3].set("net.excessiveBlock=2000000")
+
+        logging.info("Cleaning up node state")
+        for n in self.nodes:
+          n.generate(10)
+  	  self.sync_blocks()
+        
+        self.sync_all()
+        # verify mempool is cleaned up on all nodes
+	mbefore = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
+	assert_equal(mbefore,[(0, 0)]*4)
+
+
+        if 1:
+  	  logging.info("Creating addresses...")
+	  addrs = [ self.nodes[0].getnewaddress() for _ in range(TEST_SIZE+1)]
+        else:
+  	  logging.info("Loading addresses...")
+          with open("wallet10kAddrs.json") as f: addrs = json.load(f)
+
+
+        if 1:  # Test not relaying a large transaction
+          wallet = self.nodes[0].listunspent()
+          wallet.sort(key=lambda x: x["amount"],reverse=True)
+
+  	  # Create a LOT of UTXOs
+          logging.info("Create lots of UTXOs...")
+          n=0
+          group = min(100, TEST_SIZE)
+          count = 0
+          for w in wallet:
+            count += 1
+            # print count, " ",
+            split_transaction(self.nodes[0], [w], addrs[n:group+n])
+            n+=group
+            if n >= len(addrs): n=0
+	  logging.info("mine blocks")
+          self.nodes[0].generate(5)  # mine all the created transactions
+	  logging.info("sync all blocks and mempools")
+	  self.sync_all()
+
+          wallet = self.nodes[0].listunspent()
+          wallet.sort(key=lambda x: x["amount"],reverse=True)
+
+          logging.info("Test not relaying a large transaction")
+
+  	  mbefore = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
+          assert_equal(mbefore,[(0, 0)]*4)  # we need the mempool to be empty to track that this one tx doesn't prop
+
+  	  (tx, vin, vout, txid) = split_transaction(self.nodes[0],wallet[0:3000],[addrs[0]],txfeePer=60)
+          logging.debug("Transaction Length is: ", len(binascii.unhexlify(tx)))
+          assert(binascii.unhexlify(tx) > 100000) # txn has to be big for the test to work
+        
+  	  mbefore = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
+          assert_equal(mbefore[1:],[(0, 0), (0, 0), (0, 0)])  # verify that the transaction did not propagate
+          assert(mbefore[0][0] > 0) # verify that the transaction is in my node
+
+          logging.info("Test a large transaction in block < 1MB")
+          largeBlock = self.nodes[0].generate(1)
+          self.sync_blocks()
+          counts = [ x.getblockcount() for x in self.nodes ]
+          latest = counts[0]
+          assert_equal(counts, [latest,latest,latest,latest]) # Verify that all nodes accepted the block, even if some of them didn't have the transaction.  They should all accept a <= 1MB block with a tx <= 1MB
+
+  	  # mafter = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
```

Comment:
I'll delete... its more just some interesting info that could be printed out by the next developer...